freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaHealthDetailsService.java

```diff
@@ -109,17 +111,29 @@ private void addUnreachableResponse(InstanceMetaData instance, HealthDetailsFree
         response.addNodeHealthDetailsFreeIpaResponses(nodeResponse);
         nodeResponse.setName(instance.getDiscoveryFQDN());
         nodeResponse.setStatus(InstanceStatus.UNREACHABLE);
+        nodeResponse.setInstanceId(instance.getInstanceId());
         nodeResponse.addIssue(issue);
     }
 
     private HealthDetailsFreeIpaResponse updateResponse(Stack stack, HealthDetailsFreeIpaResponse response) {
         response.setEnvironmentCrn(stack.getEnvironmentCrn());
         response.setCrn(stack.getResourceCrn());
         response.setName(stack.getName());
-        if (isOverallHealthy(response)) {
-            response.setStatus(DetailedStackStatus.PROVISIONED.getStatus());
+
+        Set<String> notTermiatedStackInstanceIds = stack.getAllInstanceMetaDataList().stream()
+                .filter(not(InstanceMetaData::isTerminated))
+                .map(InstanceMetaData::getInstanceId)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+        List<InstanceStatus> nonTerminatedStatuses = response.getNodeHealthDetails().stream()
+                .filter(nodeHealthDetails -> notTermiatedStackInstanceIds.contains(nodeHealthDetails.getInstanceId()))
+                .map(NodeHealthDetails::getStatus)
+                .collect(Collectors.toList());
+        if (!nonTerminatedStatuses.isEmpty() &&
+                areAllStatusTheSame(nonTerminatedStatuses) && !hasMissingStatus(nonTerminatedStatuses, notTermiatedStackInstanceIds)) {
```

Comment:
we have 3 checks here, either of them fails we will set the status to unhealthy. Will it be obvious from the logs while we set it to unhealthy or shall we add some extra log here?
I'm just worried we will have to dig deep why it's unhealthy, but maybe it's covered by other logs.